packages/mask/src/plugins/Trader/trader/0x/useTradeCallback.ts

```diff
@@ -40,9 +40,26 @@ export function useTradeCallback(tradeComputed: TradeComputed<SwapQuoteResponse>
             type: TransactionStateType.WAIT_FOR_CONFIRMING,
         })
 
+        const config_ = {
+            ...config,
+            gas: await web3.eth
+                .estimateGas({
+                    from: account,
+                    ...pick(tradeComputed.trade_, ['to', 'data', 'value']),
+                })
+                .catch((error) => {
+                    console.log(error)
```

Comment:
If you need a more detailed error message you can use `await web3.eth.call(config_)`.